src/read_nissa_correlation.cpp

```diff
@@ -0,0 +1,204 @@
+#include<Rcpp.h>
+#include<numeric>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+//#include<bits/stdc++.h>
+
+using namespace Rcpp;
+using namespace std;
+
+//Todo: Comments
+//      Warning for not consistent order
+inline void readFile(std::istream &is,double Nt,vector<double> &real,vector<double> &imag)
+{
+  for(int i=0;i<Nt;++i)
+  {
+    double number_real;
+    double number_imag;
+    is>>number_real;
+    is>>number_imag;
+    real[i]=number_real;
+    imag[i]=number_imag;
+  }
+}
+
+//[[Rcpp::export]]
+
+
+NumericVector reading_nissa_corr(CharacterVector spinstructures,int confstart,int confnumber,int confsteps,int Nt,std::string dirname,int nsmearing,int nmasses,IntegerVector nmasses1,IntegerVector nmasses2,IntegerVector r1,IntegerVector r2){
+
+
+//SEXP reading_nissa_corr(SEXP spinstructures,SEXP confstart,SEXP confnumber,SEXP confsteps,SEXP Nt,SEXP dirname,SEXP nsmearing,SEXP nmasses,SEXP nmasses1,SEXP nmasses2,SEXP r1,SEXP r2){
+   std::filebuf fb;
+   int tindex;
+   int confindex;
+   int r1idx,r2idx;
+   int N_muindex1,N_muindex2;
+   char filename[100];
+   CharacterVector gamma_names(61);
+   NumericVector gammastructures(61);
+   int sizem1=nmasses1.size();
+   int sizem2=nmasses2.size();
+   int sizer1=r1.size();
+   int sizer2=r2.size();
+   if ((sizem1 != sizem2) || (sizem1 != sizer1) || (sizem1 != sizer2)){
```

Comment:
surely having ```sizem1 == sizer1``` is wrong, I don't think this check does what you intended, right?

Reply:
I wrote it in a way, that you can separately read in every r values for any given m. So I assume an input like (m1,r1,m2,r2). In this way you can decide for each m the r value you are interested in separately, Of coarse in practice we are interested in only ++, or --.

```suggestion
   if ((sizem1 != sizem2) || (sizem1 != sizer1) || (sizem1 != sizer2)){
```

Reply:
that's not what I mean, you're comparing ```sizem1``` with ```sizer1``` and your suggested change is exactly what you have already...

in practice we can also be interested in the off-diagonal combinations for flavour-singlets

Reply:
Sorry, I just by fall clicked on this suggested change, what I wanted to check here is just really that all four arrays (m1,m2,r1,r2) have the same length.  More precisely I have not done the direct product of the parameters in the code, but required it for the input parameters.